### PR TITLE
Improve sandbox

### PR DESCRIPTION
**Pull request by JavierSegoviaCordoba**

### CODE REVIEW

**Review by raulraja — APPROVED** [review]
All changes make sense, but we should eliminate any margin-related utilities. The Template compiler does not care about margins or indents.

**Comment by JavierSegoviaCordoba** [comment]
> All changes make sense, but we should eliminate any margin-related utilities. The Template compiler does not care about margins or indents.

Yeah, I added it because the `language=kotlin` feature from IntelliJ was showing the margins as something that doesn't compile. Not sure if we can just use `trimIndent()`, what do you think? I remember you changed it to `trimMargin()`, but I don't remember if there was a reason or just for readability.

File: arrow-reflect-annotations/src/main/kotlin/arrow/meta/samples/Product.kt [hunk 1]
```diff
@@ -14,12 +15,11 @@ annotation class Product {
...
         "product" to {
           //language=kotlin
-          """|
-             |fun product(): List<Pair<String, *>> = 
-             |  listOf(${propertiesOf(firClass) { """"${+it.name}" to this.${+it.name}""" }})
-          """.trimMargin()
+          """
+             fun product(): List<Pair<String, *>> = 
+               listOf(${propertiesOf(firClass) { """"${+it.name}" to this.${+it.name}""" }})
+          """.autoMargin()
```
[the remark this answers is not in the document]
Comment (JavierSegoviaCordoba) [reply arrow-reflect-annotations/src/main/kotlin/arrow/meta/samples/Product.kt]: Woops I didn't read this comment, I have updated changing it to use `trimIndent`, I am going to remove `trimIndent()` too.

**Comment by raulraja** [comment]
I modified the `compile` function to depend on the `@Language` annotation and automatically completes any kotlin code you add as a string in the argument. We need to generate other functions like that for call, function, class, etc...

**Pull request merged**